doc/api/crypto.md

```diff
@@ -2800,6 +2800,46 @@ threadpool request. To minimize threadpool task length variation, partition
 large `randomFill` requests when doing so as part of fulfilling a client
 request.
 
+### `crypto.randomInt([min, ]max[, callback])`
+<!-- YAML
+added:
+  - v14.8.0
+-->
+
+* `min` {integer} Start of random range. **Default**: `0`.
+* `max` {integer} End of random range (non-inclusive).
+* `callback` {Function} `function(err, n) {}`.
+
+Return a random integer `n` such that `min <= n < max`.  This
+implementation avoids [modulo
+bias](https://en.wikipedia.org/wiki/Fisher%E2%80%93Yates_shuffle#Modulo_bias).
+
+The maximum supported range value (`max - min`) is `2^48 - 1`.
+
+If the `callback` function is not provided, the random integer is generated
+synchronously.
+
+```js
+// Asynchronous
+crypto.randomInt(3, (err, n) => {
+  if (err) throw err;
+  console.log(`Random number chosen from (0, 1, 2): ${n}`);
+});
+```
+
+```js
+// Synchronous
+const n = crypto.randomInt(3);
+console.log(`Random number chosen from (0, 1, 2): ${n}`);
+```
+
+```js
+crypto.randomInt(1, 7, (err, n) => {
```

Comment:
Personal opinion: This is why I am not a huge fan of allowing a `min` argument, which is inclusive, and a `max` argument, which is exclusive (so the maximum value is `max - 1`). Users will absolutely make the mistake to think that `max` is inclusive (so the maximum value is `max`), because `min` is inclusive, and will wonder why `randomInt(1, 6)` never gives a six. Or maybe the second parameter is the number of choices, so maybe the maximum value is `min + max - 1`? I guess all that confusion is probably the reason why Java's `Random.nextInt` method still doesn't allow lower bounds.

Reply:
Would it help if the multi-parameter version accepted not `min, max`, but an object with named parameters?
Like:
`crypto.randomInt(7)`, `crypto.randomInt({ inclusiveMinimum: 1, exclusiveMaximum: 10 })`?

Reply:
Fwiw, generally the terms “minimum” and “maximum” are considered to be inclusive. However, I don’t know of a good term for the upper bound that would be implicitly *exclusive*, so I hadn’t commented before here. `start` and `end` might be better, comparable to the behavior for e.g. array functions like `.slice()`.

Reply:
@addaleax The naming could be made explicit in the named parameters version.

Moreover, I don't think that it _must_ match the single argument behavior, so perhaps we can make both bounds exclusive in this variant?

Reply:
For integers, I think the most common expectation from users would be that `max` is inclusive. For instance, if I want a coin flip, I would expect `crypto.randomInt(0, 1)` way more than I would expect `crypto.randomInt(0, 2)`.

That said, I'm becoming increasingly in favor of using the `options` object API pattern to give us the most flexibility here. e.g. `crypto.randomInt({ min: 0, max: 1}). If we did want to add `maxExclusive` later then we could do so very easily.

Reply:
Latest push uses  inclusive min and max (same as [python's randint](https://docs.python.org/3/library/random.html#random.randint)).